lib/spack/spack/directory_layout.py

```diff
@@ -150,24 +151,31 @@ def remove_install_directory(self, spec):
 
 
 class YamlDirectoryLayout(DirectoryLayout):
-    """Lays out installation directories like this::
+    """By default lays out installation directories like this::
            <install root>/
                <platform-os-target>/
                    <compiler>-<compiler version>/
-                       <name>-<version>-<variants>-<hash>
+                       <name>-<version>-<hash>
 
        The hash here is a SHA-1 hash for the full DAG plus the build
        spec.  TODO: implement the build spec.
 
-       To avoid special characters (like ~) in the directory name,
-       only enabled variants are included in the install path.
-       Disabled variants are omitted.
+       The installation directory scheme can be modified with the
+       arguments hash_len and path_scheme.
     """
 
     def __init__(self, root, **kwargs):
         super(YamlDirectoryLayout, self).__init__(root)
         self.metadata_dir   = kwargs.get('metadata_dir', '.spack')
-        self.hash_len       = kwargs.get('hash_len', None)
+        self.hash_len       = kwargs.get('hash_len')
+        self.path_scheme    = kwargs.get('path_scheme') or (
+            "${ARCHITECTURE}/${COMPILERNAME}-${COMPILERVER}/${PACKAGE}-${VERSION}-${HASH}")  # NOQA: ignore=E501
```

Comment:
It was recently discovered that ```NOQA: ignore``` ends up ignoring all flake errors. It seems the correct way to ignore the specific error is ```# NOQA: E501```